### DISCUSSION TITLE
How to override any existing visual mapping when you are into a luasnip snippet

### DISCUSSION THREAD

**Discussion by danielo515** [L3MON4D3/LuaSnip#561] [Q&A] [answered]
Hello everybody

When you expand an snippet that has jump-indexes (specially if they have a default value) lua-snip enters visual mode, with the current word selected so you can easily override it. But, this has a problem, and that is you can trigger any common visual mode mapping you have (in my case, I have one for pasting, another one for expanding the visual selection etc). This is very problematic and it is very easy to overlook.
It will be very cool if some guidance or examples (or even some sensible defaults) were given in this regard.

### Steps to reproduce

1. Expand any snippet that has jump-indexes with default values
2. just after you expand the snippet and before anything else tap any key-binding you have in the visual mode and see how badly it messes up with your snippet

**Comment by L3MON4D3** [comment, marked as answer]
Oh, the mode is actually SELECT not VISUAL.
And luckily, mappings can be defined exclusively for VISUAL, so they won't work in SELECT (see `:h mapmode-x`)

**Reply by danielo515** [reply]
Yep, I came to that conclusion when I saw some X within my existing mappings. Too lame the visual mappings also targets select mode by default
